### PR TITLE
adding tilemap game example

### PR DESCRIPTION
**Pull request by FoamyGuy**
I've added a example showing how to make a basic tile based game using displayio for PyBadge / PyGamer. I tried to knock out some PyLint stuff ahead of time but I'm sure I've still got work to do, making a PR to see where it stands.

### CODE REVIEW

File: examples/pybadger_tilegame.py [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+            # set previos button values for next iteration
+            prev_up = cur_up
+            prev_down = cur_down
+            prev_right = cur_right
+            prev_left = cur_left
+
+            # current time
+            now = time.monotonic()
+
+            # if it has been long enough based on FPS delay
```
Comment (tannewt) [inline examples/pybadger_tilegame.py:486]: Have you tried using displayio's manual refresh mode by turning auto_refresh off? It's `refresh` takes in a target frame rate which will slow down if the code is too fast and skip the frame if the code was too slow.

Comment (FoamyGuy) [reply examples/pybadger_tilegame.py:486]: Oh, neat. I did not know about that. Thank you.

**Comment by kattni** [comment]
@FoamyGuy I'm not sure this is the place for this example. I'm not sure what would be the place. Usually the examples we include expect that you're using the library in the example.

**Comment by FoamyGuy** [comment]
@kattni it does use the pybadger library for handling the button states. Though that is a relatively minor role within this example. This is one of the examples that I discussed with you and pt. I am happy to move it where every is appropriate. I was not sure if this was the exact right place either, but  I wanted to get a peek at how bad the pylint was looking, and I'm not sure how to get my local one configured the same way.

**Comment by makermelissa** [comment]
@kattni, I think he's only making use of the button reading. Perhaps creating a new library that makes reading the buttons on multiple devices with that as an example might be a better solution.

**Comment by FoamyGuy** [comment]
Partially I chose this library because it supports the same devices that this example was made for, I thought at the time at least. Though now I understand this library is getting expanded beyond pybadge and pygamer :) so I do think it's not as good of a fit with that knowledge. This example won't really support CLUE since it's using the d-pad for movement and has a different sized screen.

Perhaps a learn guide or somewhere more related to displayio might make sense though it would be nice if it's somewhere that is targeted at people using these same devices already.

**Comment by kattni** [comment]
@FoamyGuy is going to test this example with the upcoming refactor of PyBadger, and submit a new PR once the refactor is merged. We discussed it, and are choosing to keep it within this repo for now, but under a deeper directory structure within `examples`.

**Pull request closed without merging**